### PR TITLE
Added the file-upload directive.

### PR DESCRIPTION
**Pull request by KamilaHasanbega**

### CODE REVIEW

File: chrome/components/fileupload/fileupload-directive.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    },
+    restrict: 'A',
+    scope: {
+      /** @export */
+      accept: '@asAccept',
+      /** @export */
+      upload: '&asUpload'
+    },
+    template: '<ng-transclude></ng-transclude>' +
+        '<input tabindex="6" id="now" type="file" style="display:none;' +
```
Comment (koto) [inline chrome/components/fileupload/fileupload-directive.js]: Why do you need the `tabindex` if the `display` is `none`? Tabbing to this input field would not work. Similarly, why is `id` needed?

File: chrome/components/fileupload/fileupload-directive.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      // Upload each file that the user has selected from the modal dialog.
+      inputAttachment.addEventListener('change', function(event) {
+        var files = event.target.files;
+
+        angular.forEach(files, function(file) {
+          var reader = new FileReader();
+          reader.onload = function(fileEvent) {
+            scope.$apply(function() {
+              scope.upload({'name': file.name, 'type': file.type,
+                'contents': unescape(
```
Comment (koto) [inline chrome/components/fileupload/fileupload-directive.js]: Please comment in the code why you're doing unescape(encodeURIComponent).

Comment (koto) [inline chrome/components/fileupload/fileupload-directive.js]: This (unescape(encodeURIComponent)) should be changed to use Uint8Arrays instead.

**Comment by koto** [comment]
LGTM, safe to merge @kbsriram.

**Comment by kbsriram** [comment]
rebased via d7845a2

**Pull request closed without merging**